### ISSUE TITLE
Support enabling strict in config with rule-specific overrides

### ISSUE THREAD

**Issue by jab** [microsoft/pyright#3198]
**Is your feature request related to a problem? Please describe.**
There is currently no way to configure Pyright as in the following example:
```json
{
  "strict": ["src"],
  "reportPrivateUsage": false
}
```
In other words, one cannot globally enable all strict checks in the `src` tree, except for one particular strict check that is not helpful (perhaps due to a Pyright bug causing a large number of false positives).

**Describe the solution you'd like**
It should be possible to combine the `strict` setting with rule-specific overrides, as in the example above.

**Additional context**
This is currently supported by other type checkers, such as mypy. Users coming to Pyright from other type checkers may experience this as extra friction in adopting Pyright.

**Comment by erictraut** [comment]
This runs counter to the design of the configuration system in pyright, so we're unlikely to implement this. The current design is consistent with the TypeScript compiler. Implementing this proposal would be complicated and confusing.

If you want to enable a subset of diagnostic rules, you can enumerate them individually. The "strict" mode is a shortcut for "turn on pretty much everything, including new diagnostic rules". 

You can use `# pyright` comments to disable specific strict-mode rules on a per-file basis if necessary.

**Comment by jab** [comment]
Oh, I’m not familiar with the typescript compiler, so coming from the world of mypy / other Python linters where this sort of thing is common, I had no idea this would be complicated or confusing for pyright.

Just in case, would something like flake8’s “per file ignores” be more doable for pyright? https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores

That would also allow centrally managing the ignores from the config file, which can be preferable over littering various pyright config across each file in the source.

**Comment by erictraut** [comment]
The configuration mechanisms in pyright are already very extensive and complex. We need to merge configuration sources settings from the LSP, command-line options, pyrightconfig.json, pyproject.toml, and per-file overrides (via comments). It's complicated to maintain, document, and explain to users. The justification for adding yet another override mechanism would need to be incredibly compelling for us to consider it. You are the first person to request this. We might reconsider this in the future if we get strong signal from a sufficient number of users, but for now this is a pretty firm no.

**Comment by jab** [comment]
Got it, thanks for considering!

**Comment by baggiponte** [comment]
@erictraut thank you for your answer. Just to leave this here, I too believe that a per-file config might be helpful: for example, it might be used in `__init__.py` to get rid of the inevitable `"..." is not accessed` message. The current alternative is to plainly ignore/exclude checking `__init__.py` files altogether.

**Comment by Julian** [comment]
Calling something out of scope is of course totally understandable, so feel free to ignore, but just wanted to share that as a (new-ish) user of pyright, I found the current behavior quite surprising too (and ultimately found this issue).

Perhaps another option to consider at some point is at least warning when a `pyproject` or config file contains simultaneously `strict` and a `report*` option, it certainly would have saved a bit of time trying to figure out why the `report*` options seemed to be silently ignored.
